altair/vegalite/v5/theme.py

```diff
@@ -60,3 +72,75 @@ def __repr__(self) -> str:
     themes.register(theme, VegaTheme(theme))
 
 themes.enable("default")
+
+
+def register_theme(
+    name: LiteralString, *, enable: bool
```

Comment:
Any specific reason to restrict to `LiteralString` here? The functionality works for any `str` and although it might be niche, users could autogenerate theme names. Maybe they would like to register multiple themes in various combinations and then create theme names with f string syntax.

Reply:
So it is related to needing the string to be literal here, as having `str` instead would "swallow" the other `Literal` aliases - preventing them from showing on hover:

https://github.com/vega/altair/blob/5207768b6e533c0509218376942309d1c7bac22f/altair/utils/theme.py#L22-L25

For the same reason I had to update the type here https://github.com/dangotbanned/altair/blob/027c82cbc469c811a18e6f914e3a3c1d8cf12a00/altair/vegalite/v5/theme.py#L31

---

FWIW I did try `str` first. 
Settling on `LiteralString` was more of a pragmatic decision on my part, trying to prioritise the UX of the most general case. 
That is, having autocomplete with the *occasional* `#type: ignore` is preferable to no autocomplete at all.

Does that additional context help at all @binste?

Reply:
Added [requested comments](https://github.com/vega/altair/pull/3526#pullrequestreview-2281646031) in [`9ce798c` (#3526)](https://github.com/vega/altair/pull/3526/commits/9ce798c15ec01c57d2d810d9215815a3a75bda92)